config/action.d/pf.conf

```diff
@@ -4,16 +4,52 @@
 #
 # Author: Nick Hilliard <nick@foobar.org>
 # Modified by: Alexander Koeppe making PF work seamless and with IPv4 and IPv6
+# Further modified by: Zane C. Bowers-Hadley <vvelox@vvelox.net> to make it kill states for a IP so the
+#                      ban takes affect as well as not handling multiple protocols.
+
+# WARNING WARNING WARNING WARNING WARNING
+# WARNING WARNING WARNING WARNING WARNING
+# WARNING WARNING WARNING WARNING WARNING
+#
+# This will kill any state for a banned IP. This will ensure the ban actually takes affect.
+# This is need as pf has both the strenght and weakness that changes won't affect current
+# states till it is forced to, but unfortunately the only way to make that take affect is
+# to kil them.
+#
+# If some one wishes to fix that, this will requiring writing a tool to grab the session
+# id for the item in question from 'pfctl -s state -vv'.
+#
+# This is especially important for stateless procols such as UDP as well as TCP protocols
+# that are capable of processing multiple requests in a single session, such as HTTP2.
+#
+# If you wish to disable this, you need to set no_kill=true with an action.
+#
+# Example..
 #
+# [dovecot]
+# enabled = true
+# filter = dovecot
+# action = pf[actiontype=<allports>,no_kill="true"]
 #
+# The above will mean if some one hits the limit for dovecot, it won't automatically kill their
+# active states, such as maybe SSH.
+#
+# Where would be doing this be bad? Lets say you have a service using UDP. The attacker can keep it
+# up indefinitely as long they don't let up as UDP is stateless. Similarly with HTTP2, a attacker
+# can keep it going till the web server kills or they finish doing what ever they are doing. This is
+# especially true for a rule that triggers based on attempted data exfiltration.
+#
+# Basially you have the choice of maybe a potetional moment annoyance of having to restart a SSH
+# session or the like or potiential compromise as the pf table change was never allowed to take affect
+# as the states for that IP were never cleared.
```

Comment:
This paragraph should be rewritten to discuss the risk of long running sessions, such as SSH. However, how could a currently active session cause a ban? Typically if a user has logged in and subsequently caused fail2ban to add the IP to the block list one has more problems than some bad actor attempting to gain access. I would think that such scenario would occur when a valid user has successfully logged in from behind a NAT firewall and another person behind the same NAT firewall has tripped fail2ban. I suppose to be safe the SSH session could be shut down.

Considering another scenario a bad actor could DoS the valid user. For example, when I travel my ISP (cable company) maintains WiFi nodes throughout Canada. I simply connect. However some bad actor could deny me access to my site because this ISP maintains a NAT appliance in each different city.

I'm leaning toward not including this paragraph at all because less experienced users could make the wrong choice because of insufficient information to make an informed decision. Or, this paragraph could be reworded but I think this will confuse the issue even more.

Reply:
This is what the no_kick bit is for.

That said a issue like you mentioned there is easily fixable via just pivoting where you are routing through post leaving the WAP. :P

You are right, that said I disagree about not including it. IMHO you make a great argument for expanding upon it, better listing the failure modes for both to help the user understand when selecting for one.